src/main/kotlin/io/emeraldpay/dshackle/rpc/NativeCall.kt

```diff
@@ -242,6 +242,12 @@ open class NativeCall(
             val requestDecorator = getRequestDecorator(requestItem.method)
             val resultDecorator = getResultDecorator(requestItem.method)
 
+            val selector = if (request.hasSelector()) {
+                Selectors.keepForwarded(request.selector)
```

Comment:
`val selector =request.takeIf {it.hasSelector()}?.let{Selectors.keepForwarded(it.selector)}`

Reply:
if you have else { return null } in your code - it's java style, not a kotlin